source/core/slang-io.cpp

```diff
@@ -777,6 +779,61 @@ namespace Slang
 #endif
     }
 
+    /* static */SlangResult Path::removeNonEmpty(const String& path)
+    {
+        if (File::exists(path) == false)
+        {
+            return SLANG_OK;
+        }
+
+        StringBuilder msgBuilder;
+        // Path::remove() doesn't support remove a non-empty directory, so we need to implement
+        // a simple function to remove the directory recursively.
+#ifdef _WIN32
+        // https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shfileoperationa
+        // Note: the fromPath requires a double-null-terminated string.
+        String newPath = path;
+        newPath.append('\0');
+        SHFILEOPSTRUCTA file_op = {
+            NULL,
+            FO_DELETE,
+            newPath.begin(),
+            "",
```

Comment:
> This parameter must be set to NULL if it is not used.

We should probably set this to `NULL`  instead of `""` to avoid any temperamental issues with WinAPI and promotion/demotion to `PCZZSTR`.

Reply:
ack.